Remove duplicated "internal" types and wrappers

LeCodeurDuDimanche opened the pull request:
Removed duplicated internal object definitions and wrappers around them : there weren't needed and this simplifies the code base. It is also more coherent and may resolve some problems around object

LeCodeurDuDimanche committed:
Remove duplicated "internal" types and wrappers

Removed duplicated internal object definitions and wrappers around them : there weren't needed and this simplifies the code base. It is also more coherent and may resolve some problems around object

davepuchyr commented on internal/store/indexes/filtering.go:
```diff
@@ -5,12 +5,13 @@ import (
 	"github.com/iov-one/cosmos-sdk-crud/internal/store/iterator"
 	"github.com/iov-one/cosmos-sdk-crud/internal/store/types"
 	"github.com/iov-one/cosmos-sdk-crud/internal/util"
+	types2 "github.com/iov-one/cosmos-sdk-crud/types"
```
Please use a more meaningful name.  The de facto standard would be to use `crudtypes`.

davepuchyr replied:
Actually, `crud` would be better because then it'd be consistent with other files.

LeCodeurDuDimanche replied:
Yeah I have to change that, it was the default refactoring name

davepuchyr commented on internal/store/indexes/store.go:
```diff
@@ -3,6 +3,8 @@ package indexes
 	"fmt"
 
+	types2 "github.com/iov-one/cosmos-sdk-crud/types"
+
```
Please drop the blank line.

davepuchyr commented on internal/store/indexes/store_test.go:
```diff
@@ -8,7 +8,8 @@ import (
 
-	"github.com/iov-one/cosmos-sdk-crud/internal/store/types"
+	types2 "github.com/iov-one/cosmos-sdk-crud/types"
+
```
Stupid goland.

davepuchyr commented on internal/store/objects/store.go:
```diff
@@ -3,6 +3,8 @@ package objects
 	"fmt"
 
+	types2 "github.com/iov-one/cosmos-sdk-crud/types"
+
```
goland

davepuchyr commented on internal/store/objects/store_test.go:
```diff
@@ -5,11 +5,12 @@ import (
 	"testing"
 
+	types2 "github.com/iov-one/cosmos-sdk-crud/types"
+
```
Why does it do this?!?

LeCodeurDuDimanche committed:
Moved types package to root for compability and fixed style

davepuchyr commented:
> Removed duplicated internal object definitions and wrappers around them : there weren't needed and this simplifies the code base. It is also more coherent and may resolve some problems around object

@fdymylja , what's the reason for wrapping the objects?

LeCodeurDuDimanche committed:
Move crud.Store out of the internal package

Now that the wrappers are gone, this is the entry point of the package, so it has to be public

fdymylja commented:
> > Removed duplicated internal object definitions and wrappers around them : there weren't needed and this simplifies the code base. It is also more coherent and may resolve some problems around object
> 
> @fdymylja , what's the reason for wrapping the objects?

Friendlier API. It was made in order to just have to import crud and have access to everything needed to interact with crud. 

It's not needed and it can be removed.
